### ISSUE TITLE
Viruses somehow broken after lcass's smoke check commit.

### ISSUE THREAD

**Issue by feotusgod** [HippieStationArchive/HippieStation13-1#419]
Viruses from virology and virus crates have no virus, and admin made viruses dont register on the computer (but works and can spread by blood) and cant receive new syntomes.

lcass's breaking commit is: https://github.com/HippieStationCode/HippieStation13/commit/dcaf17f6499e72e30392cae01812b591347bec73

This is the version where viruses work: https://github.com/HippieStationCode/HippieStation13/tree/8a6a8dcd6a236063b3f3652f4829ead8c8b58e0c

This is an error that you get when mixing mutagen/plasma/synaptizene with blood without a virus (made by an admin):

```
runtime error: Undefined operation: /list (/list) * 0.333333
proc name: add reagent (/datum/reagents/proc/add_reagent)
  source file: Chemistry-Holder.dm,405
  usr: Admin (/mob/living/carbon/human/dummy)
  src: /datum/reagents (/datum/reagents)
  call stack:
/datum/reagents (/datum/reagents): add reagent("mutagen", 10, 270, null)
/datum/reagents (/datum/reagents): trans to(Rhinovirus culture bottle (/obj/item/weapon/reagent_containers/glass/bottle/cold), 10, 1, 1)
the unstable mutagen bottle (/obj/item/weapon/reagent_containers/glass/bottle/mutagen): afterattack(Rhinovirus culture bottle (/obj/item/weapon/reagent_containers/glass/bottle/cold), Admin (/mob/living/carbon/human/dummy), 1, "icon-x=14;icon-y=11;left=1;scr...")
Admin (/mob/living/carbon/human/dummy): ClickOn(Rhinovirus culture bottle (/obj/item/weapon/reagent_containers/glass/bottle/cold), "icon-x=14;icon-y=11;left=1;scr...")
Rhinovirus culture bottle (/obj/item/weapon/reagent_containers/glass/bottle/cold): Click(the floor (157,101,1) (/turf/simulated/floor/plasteel), "mapwindow.map", "icon-x=14;icon-y=11;left=1;scr...")

```

It points to this (but i dont know how this relates to why viruses that start in bottles don't work):

```
datum/reagents/proc/add_reagent(var/reagent, var/amount,var/temp = 270, var/list/data=null)
    if(!isnum(amount)) return 1
    update_total()
    if(total_volume + amount > maximum_volume) amount = (maximum_volume - total_volume) //Doesnt fit in. Make it disappear. Shouldnt happen. Will happen.
    //adjust temperature
    if(total_volume > 0)
        var/mult = amount/(total_volume + amount)
        var/temp_dif = present_machines[1] - temp//assume it's 270 , I can't be bothered to change all the add_reagent procs atm
(405)->     present_machines[1] -= temp_dif * mult// apply the new temperature to the current reagent holder
//more code here...
```

The real weird thing is that in the broken virus version, i removed the commit manually, and recompiled it but it still wont work. Then i did a "winmerge" to see any differnces just in case, and only the binaries are different (no readable text other than tgstation.int which is just the build time).

**Comment by Jaygriff** [comment]
DO other viruses work?

**Comment by feotusgod** [comment]
No. The viruses in the virus crate are also plain blood. 
Injecting the plain blood does not give the virus.

Apparently in the abductor gamemode the ayyliums can get viruses.

**Comment by lcass** [comment]
Hmmm , I'll take a look it might be something to do with that.

**Comment by Wherix** [comment]
To add onto this I have some chat logs of some of the odd interactions between the virus synthblood and containers. I talked to Chronitonity and he brought me a viro crate to test.

The album of pictures is here: http://imgur.com/a/C1OjA

Some other things to note:
1. Not a single vial of virus came up on the pandemic machine.
2. Mixed viruses never show up on the pandemic machine either, and it seems like any way you mix them excluding syringes (and presumably the dripper), you only ever get 10 units no matter how much you added of each one. When using a syringe the total is 5, and for the dripper I'd have to assume the total would be 1. (edit: I realised this probably has to do with the transfer rates on whatever containers you use)
3. I played Virology yesterday August 19th in a time frame of about 8 - 12 pm central time, and Virology was working perfectly. Then when I logged in today August 20th it was broken again.

Hope some of this information helps you guys fix this.
